src/main/java/io/pinecone/clients/Index.java

```diff
@@ -205,6 +209,7 @@ public DescribeIndexStatsResponse describeIndexStats(Struct filter) {
     public void close() {
         try {
             logger.debug("closing channel");
+            pinecone.getConnectionsMap().remove(indexName);
```

Comment:
remove the connection from the connections map here

Reply:
Good point. 
1. Since pinecone object is instantiated in the Pinecone class and the constructors new Index() + new AsyncIndex() are called within Pinecone class, they are not going to be null. But I have added the null check for pinecone object in the constructor. @ssmith-pc thoughts?
2. If map.get(indexName) is not found, map.remove won't delete the connection. So the checks are not required on the indexName.

Reply:
@ssmith-pc another option I was thinking for closeConnection(indexName) is making it static, so I can call the method from close() of Index and AsyncIndex classes with class Pinecone instead of checking if the pinecone object is available or not.

Reply:
> If map.get(indexName) is not found, map.remove won't delete the connection. So the checks are not required on the indexName.
Check the Conc Hash map docs to confirm, but I think `CHM.remove(null)` will throw an NPE. There are a few other cases in the API where the client is expected to pass in the name of an index and I don't see it being validated anywhere that's a non-empty string. Would be a good idea to do some basic input validation before sending to server.

> another option I was thinking for closeConnection(indexName) is making it static
Is it true that users are never expected to call this, it's just an internal method to close the connection? I think that would be ok to make it static since it only interacts with the static CHM.

Reply:
1. That's correct, CHM.remove(null) throws NPE since it calls containsKey() first.
CHM.remove(key) where key doesn't exists in the map results in no action. 

2. Yes, users are not expected to interact with closeConnection(indexName).